development/CONVENTIONS.md

```diff
@@ -0,0 +1,127 @@
+# Conventions for the project _helpwave_
+
+## Content
+- [Introduction](#introduction)
+- [Git](#git)
+  - [Branches](#branches)
+  - [Commit messages](#commit-messages)
+  - [Merge commits](#merge-commits)
+
+## Introduction
+This convention is a set of rules to be followed by all developers working on the project and repositories of _helpwave_ to help
+maintain a consistent code style.
+
+## Git
+### Branches
+The project uses the following branches:
+* `main` - the main branch, contains the latest stable version of the project
+* `issue/<issue-number>-<issue-name>` - a branch for a specific issue, where `<issue-number>` is the number of the issue
+and `<issue-name>` is the name of the issue.
+
+### Commit messages
+The projects use the [Conventional Commits](http://conventionalcommits.org) scheme.
+Commit messages should be written in the following format:
+```text
+<type>[(<scope>)]: <message>
+```
+where `<type>` is the type of the commit, `<message>` is the message of
+the commit and `<scope>` represents an optional sub-part of the type.
+
+#### Type
+The type of the commit message should be one of the following:
+* `feat` - a new feature
+* `fix` - a bug fix
+* `docs` - changes to the documentation
+* `style` - formatting, missing semicolons, etc.; no code change
+* `refactor` - refactoring production code
+* `test` - adding tests, refactoring test; no production code change
+* `chore` - updating build tasks, package manager configs, etc.; no production code change
+* `ci` - changes to the CI configuration files and scripts
+* `wip` - work in progress
+* `infra`
+
+#### Message
+The message of the commit should be written in the imperative, present tense: "change" not "changed" nor "changes". The
+first letter should not be capitalized. The message should **not** contain a dot (.) at the end.
+The maximum length of the message is 50 characters and the language should be english.
+
+#### Scope
+The scope represents a sub-part of the project, like a specific microservice.
+
+#### Examples
+```text
+feat: add the ability to create a new user
+```
+```text
+fix: fix the bug with the user creation
+```
+```text
+docs: add the documentation for the user creation
+```
+```text
+fix(infra): reduce costs
+```
+
+## Merge commits
+Merge commits from a feature branch to the main branch should be in the following format:
+```text
+Merge from 'issue/<issue-number>-<issue-name>' into main (#<pull-request-number>)
+```
+where `<issue-number>` is the number of the issue, `<issue-name>` is the name of the issue and `<pull-request-number>`
+is the number of the pull request.
+### Examples
+```text
+Merge from issue/1-add-the-ability-to-create-a-new-user into main (#1)
+```
+```text
+Merge from issue/2-fix-the-bug-with-the-user-creation into main (#2)
+```
+[…]
```

Comment:
```suggestion
Merge commits should look like what github suggests in a pull request.
```

Reply:
For example, the `services` repository prefers squashed merge commits. With that in mind, the commit message should be valid against our commit message convention.

I accept the suggestion for now, we didn't enforce this policy yet.

Reply:
Everyone is able to fulfill this requirement while using git squashing. I don't see any reason not implementing this style.